src/components/gamespec/Board.js

```diff
@@ -5,18 +5,54 @@ import ItemTypes from './ItemTypes';
 import { Layer, Stage } from 'react-konva';
 
 import CanvasImage from './CanvasImage';
+import { piecesRef } from '../../firebase';
 
 const boxTarget = {
   drop(props, monitor, component) {
       let offset = monitor.getClientOffset(),
           item = monitor.getItem();
 
       let items = props.getItems();
+      let pieces = props.getPieces();
       let rect = component.refs.stage.getStage().getContainer().getBoundingClientRect();
       offset.x = offset.x - rect.left;
       offset.y = offset.y - rect.top;
+      let images = [];
+      items.forEach((element, index) =>{
+        let x = element.offset.x - offset.x
+        let y = element.offset.y - offset.y
+        if (Math.pow(x,2) < Math.pow(element.image.width,2) && Math.pow(y,2) < Math.pow(element.image.height,2)) {
+          pieces.forEach((piece, index) =>{
+            if(piece.childKey === element.childKey){
+              images = piece.images;
+              let flag = 0;
+              images.forEach((pic, index) =>{
+                if(pic.key === item.image.key){
```

Comment:
This is probably wrong. A person can drag an image multiple times on board, so you don't need to compare it like this.

Reply:
We are confused that if we only have pieces and delete items, how should we deal with drag action since the index of each item is not absolute any more?

Reply:
You do multiple for loops as I explained above. Also, you can implement in any way you want, I am just giving possible suggestions. If you want to keep multiple pieces you can go with it.

Furthermore, for unique index you can think of piece and images as a linear array.
Let's say you have two pieces, one with 3 images and other with 2 images.
So the index of each image becomes uniquely as 0, 1, 2, 3, 4 where you can actually flatten the array. There are multiple ways to do this, this is just one of them. You are however, free to implement the functionality however you want as long the final working functionality is the one we expect.